pkg/utils/v1alpha1/kubernetes.go

```diff
@@ -86,6 +86,20 @@ func CreateCSIVolumeCR(csivol *apis.CSIVolume, nodeID, mountPath string) (err er
 	return
 }
 
+// UpdateCSIVolumeCR updates CSIVolume CR related to current nodeID
+func UpdateCSIVolumeCR(csivol *apis.CSIVolume) error {
+
+	oldcsivol, err := csivolume.NewKubeclient().WithNamespace(OpenEBSNamespace).Get(csivol.Name, v1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	oldcsivol.Spec.Volume.DevicePath = csivol.Spec.Volume.DevicePath
+	oldcsivol.Status = csivol.Status
+
+	_, err = csivolume.NewKubeclient().WithNamespace(OpenEBSNamespace).Update(oldcsivol)
```

Comment:
Why are we instantiating the kubeclient multiple times?
Why dont we set this kubeclient as a field of some struct.
This also indicates the need to move the logic from a function to a method.

Reply:
Let's discuss this further. I will create a separate PR for the same, as this will need to be refactored at multiple places.